pkg/netutil/netutil.go

```diff
@@ -44,6 +48,51 @@ type CNIEnv struct {
 
 type CNIEnvOpt func(e *CNIEnv) error
 
+func UsedNetworks(ctx context.Context, containers []containerd.Container) (map[string][]string, error) {
+	used := make(map[string][]string)
+	for _, c := range containers {
+		task, err := c.Task(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		status, err := task.Status(ctx)
+		if err != nil {
+			return nil, err
+		}
+		switch status.Status {
+		case containerd.Paused, containerd.Running:
+		default:
+			continue
+		}
```

Comment:
I think this entire logic should move to lsutil in getContainersInAllNamespaces(..., []excludeContainerStatus) which is a new parameter; so that netutil does not need to care about containers lifecycle.

Reply:
make sense, thanks

Reply:
Done